tools/util/ReplayOperationManager.cpp

```diff
@@ -907,21 +927,21 @@ void ReplayOperationManager::makeAllocateCont( uint64_t allocation_from_log )
 
 void ReplayOperationManager::makeDeallocate( int allocator_num, uint64_t allocation_from_log )
 {
-  m_cont_op = new ReplayOperation(m_allocator_array, m_alloc_operations);
+  m_cont_op = new ReplayOperation(m_allocator_array, m_alloc_operations, m_resource_names);
   m_cont_op->makeDeallocate(allocator_num, allocation_from_log);
   operations.push_back(m_cont_op);
 }
 
 void ReplayOperationManager::makeCoalesce( const std::string& allocator_name )
 {
-  m_cont_op = new ReplayOperation(m_allocator_array, m_alloc_operations);
+  m_cont_op = new ReplayOperation(m_allocator_array, m_alloc_operations, m_resource_names);
   m_cont_op->makeCoalesce(allocator_name);
   operations.push_back(m_cont_op);
 }
 
 void ReplayOperationManager::makeRelease( int allocator_num )
 {
-  m_cont_op = new ReplayOperation(m_allocator_array, m_alloc_operations);
+  m_cont_op = new ReplayOperation(m_allocator_array, m_alloc_operations, m_resource_names);
```

Comment:
There must be a better way of doing this.  I suppose what I should do is create a separate ReplayOperationEnvironment class for environment information set up by the ReplayOperationManager and then shared/utilized by each of the ReplayOperation objects.  I could then pass along a reference to the ReplayOperationEnvironment object to the operations.  Is this a pattern?

Reply:
@davidbeckingsale, As discussed, I will be modifying replay to pass a reference to the replayManager to all of the replayOperations.

Reply:
At a later date, right?

Reply:
Oops.